zubhub_frontend/zubhub/src/components/form/tagsInput/TagsInput.jsx

```diff
@@ -28,66 +28,107 @@ export default function TagsInput({
   const commonClasses = makeStyles(styles)();
   const classes = makeStyles(tagsInputStyles)();
   const [isSearching, setIsSearching] = useState(false);
-  selectedTags = [...(selectedTags ?? [])];
+  const [errorMessage, setErrorMessage] = useState('');
+  const selectedTags = [...(initialSelectedTags ?? [])];
   const ref = useRef(null);
 
-  const handleChange = e => {
-    setIsSearching(true);
-    onChange(e.target.value);
-  };
+  const validateTag = useCallback((tag, selectedTags) => {
+    const trimmedTag = tag.trim();
+    if (trimmedTag === '') {
+      return ['Tag cannot be empty.', false];
+    } else if (selectedTags.includes(trimmedTag)) {
+      return [`The tag "${trimmedTag}" is already in the search bar and cannot be added again.`, false];
+    } else {
+      return ['', true];
+    }
+  }, []);
 
-  const handleClickAway = () => setIsSearching(false);
-  const handleFocus = () => {
-    clearSuggestions();
-  };
+  const handleTagAddition = useCallback((value) => {
+    const [errMsg, isValid] = validateTag(value, selectedTags);
+    setErrorMessage(errMsg);
+    if (isValid) {
+      addTag(value);
+      setIsSearching(false);
+      setErrorMessage('');
+    }
+  }, [addTag, selectedTags, validateTag]);
 
-  const handleTagAddition = value => {
-    addTag(value);
-    setIsSearching(false);
-  };
+  const handleChange = useCallback((e) => {
+    const inputValue = e.target.value.trim();
+    const [errMsg, isValid] = validateTag(inputValue, selectedTags);
+    setIsSearching(isValid);
+    setErrorMessage(errMsg);
+    if (isValid) {
+      onChange(inputValue);
+    }
+  }, [onChange, selectedTags, validateTag]);
 
-  const handleKeyDown = e => {
+  const handleKeyDown = useCallback((e) => {
     if (e.key === 'Enter') {
-      addTag(e.target.value);
+      const inputValue = ref.current.value.trim();
+      const [errMsg, isValid] = validateTag(inputValue, selectedTags);
+      setErrorMessage(errMsg);
+      if (isValid) {
+        addTag(inputValue);
+        setIsSearching(false);
+        setErrorMessage('');
+        ref.current.value = '';
+      }
+    }
+  }, [addTag, selectedTags, validateTag]);
+
+  const handleClickAway = useCallback(() => {
+    const inputValue = ref.current.value.trim();
+    if (!inputValue || selectedTags.includes(inputValue)) {
       setIsSearching(false);
+      setErrorMessage('');
+    } else {
+      const isFound = remoteData.some(tag => tag.name === inputValue);
+      if (isFound || !remoteData.length) {
+        addTag(inputValue);
+        setIsSearching(false);
+        setErrorMessage('');
+        ref.current.value = '';
+      } else {
+        setErrorMessage(`The tag "${inputValue}" was not found and cannot be added.`);
```

Comment:
missing translation

Reply:
As for this the translations where not part of this issue, so to be moved to the next PR :)